src/lightning/store/store.py

```diff
@@ -0,0 +1,89 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from typing import List
+
+from lightning_cloud.openapi import V1Model, V1UploadModelRequest
+
+from lightning.app.utilities.cloud import _get_project
+from lightning.store.utils import _Client, _download_file_from_url, _upload_file_to_url
+
+
+def upload_model(
+    name: str,
+    path: str,
+    version: str = "latest",
+    progress_bar: bool = True,
+) -> None:
+    """Upload a model to the lightning cloud.
+
+    Args:
+        name:
+            The model name.
+        path:
+            The path to the checkpoint to be uploaded.
+        version:
+            The version of the model to be uploaded. If not provided, default will be latest (not overridden).
+        progress_bar:
+            A progress bar to show the uploading status. Disable this if not needed, by setting to `False`.
+    """
+    client = _Client()
+    user = client.auth_service_get_user()
+    # TODO: Allow passing this
```

Comment:
are you planning to resolve this in this PR or a follow-up? This seems somewhat essential.

Reply:
Yeah, will get to it in a follow-up